core/src/openid.cpp

```diff
@@ -155,6 +155,15 @@ namespace irods::http::openid
 				return std::nullopt;
 			}
 		}
+		// Some IAM servers (e.g. keycloak) could be set up
+		// to exclude `aud' from a bearer token payload
+		// If no 'aud' was found in bearer token, do not accept
+		else if (auto strict_aud{irods::http::globals::oidc_configuration().find("strict_introspection_endpoint_aud")};
+		         strict_aud != std::end(irods::http::globals::oidc_configuration()) && strict_aud->get<bool>())
```

Comment:
Does this logic hold if the config property isn't defined?

Reply:
If the OIDC stanza doesn't exist? Or if `strict_introspection_endpoint_aud` doesn't exist?

Reply:
If `strict_introspection_endpoint_aud` doesn't exist.

If the OIDC config stanza doesn't exist, we never reach this point AFAIK.

Reply:
This should work. We check to see if exists via `.find(...)`. If we did find something, then we can check the value. Having the option missing treats this as an implicit 'false'.

Basically, the option is opt-in. It required both the config option to be defined, and the right value ('true').